### ISSUE TITLE
Wrong DB used when Distribute table join Local table

### ISSUE THREAD

**Issue by ljr900723** [ClickHouse/ClickHouse#10471]
There is a DB named "jrliu_test" I created before.
Then I connect to Clickhouse like this:
`clickhouse-client -m --database=jrliu_test`

Then I execute the followint SQLs:
1. create table "test_table_local"
`CREATE TABLE IF NOT EXISTS jrliu_test.test_table_local ON CLUSTER '{cluster}' (
	join_key String,
	value String
) ENGINE = MergeTree()
PARTITION BY tuple()
ORDER BY tuple();`

2. create table "test_table_distribute"
`CREATE TABLE jrliu_test.test_table_distribute ON CLUSTER '{cluster}' AS jrliu_test.test_table_local ENGINE = Distributed('{cluster}', 'jrliu_test', test_table_local);`

3. create table "test_table_join"
`CREATE TABLE IF NOT EXISTS jrliu_test.test_table_join ON CLUSTER '{cluster}' (
	join_key String,
	value String
) ENGINE = MergeTree()
PARTITION BY tuple()
ORDER BY tuple();`

4. insert data to "test_table_local" and "test_table_join"
`INSERT INTO jrliu_test.test_table_local FORMAT Values ('key1', 'value1'), ('key2', 'value2');`
`INSERT INTO jrliu_test.test_table_join FORMAT Values ('key1', 'value1'), ('key3', 'value3');`

5. execute query
5.1 execute Local table join Local table
`SELECT * FROM test_table_local as t INNER JOIN test_table_join as j ON t.join_key = j.join_key;`
(This query return result without any errors)
5.2 execute Distribute table join Local table
`SELECT * FROM test_table_distribute as t INNER JOIN test_table_join as j ON t.join_key = j.join_key;`
(This query return error: **_Code: 60, e.displayText() = DB::Exception: Received from clickhouse-stg-af-shard1-replica0:9000. DB::Exception: Table default.test_table_join doesn't exist.. (version 19.14.7.15 (official build))_**)

Table "test_table_join" should be found in "jrliu_test" database but not in "default". 

Server version: 19.14.7.15

**Comment by 4ertus2** [comment]
It's most probable duplicate for #4756
It should be fixed in 20.4

**Comment by azat** [comment]
AFAICS the problem here is different, distributed engine do not account current database, IOW:

```sql
use system;
create table dist_parts as parts engine=Distributed(test_cluster_two_shards, currentDatabase(), parts);
select * from system.dist_parts l join parts r using name format Null;
```

Will trigger the following error:

```
Code: 60. DB::Exception: Received from localhost:9000. DB::Exception: Received from 127.0.0.2:9000. DB::Exception: Table default.parts doesn't exist.. 
```

But note, that the error is only on the remote shard, on the local shard it is ok

**Comment by 4ertus2** [comment]
Could be related #15538

**Comment by dzeban** [comment]
Still reproduced on the latest stable version 20.10.3.30, so fix in #15538 doesn't help here.
